linter/linter.go

```diff
@@ -147,18 +151,23 @@ func Run() bool {
 	_, graphResults := gr.CheckSequences()
 	seqResults.Union(graphResults)
 	if seqResults.AnyError {
+		errorPrinted := false // whether we printed anything
 		for _, seq := range sequences {
 			header, err := fmtHeader(seq, allSpecs)
 			if err != nil {
 				fmt.Fprintf(os.Stderr, "%s", err)
 			}
-			linter.printCheckResult(header, seqResults.Results[seq])
+			errorPrinted = linter.printCheckResult(header, seqResults.Results[seq]) || errorPrinted
+		}
+		if errorPrinted {
+			return false
 		}
-		return false
 	}
 
-	// 5. No errors occurred: print warnings, then exit.
+	// 5. No errors occurred, and we've printed all warnings. Decide whether
```

Comment:
"we've printed all warnings"  but there's still warnings to print

Reply:
-_-